clang build failing on Travis

The Travis clang builds are suddenly failing.  Maybe the clang version was
upgraded on Travis.

https://travis-ci.org/DynamoRIO/dynamorio/jobs/273692624
```
debug-internal-32: **** 4 build errors ****
	/.../dynamorio/suite/tests/linux/signal_racesys.c:104:28: error: implicit conversion from 'long long' to '__time_t' (aka 'long') changes value from 3155760000 to -1139207296 [-Werror,-Wconstant-conversion]
	make[2]: *** [suite/tests/CMakeFiles/linux.signal_racesys.dir/linux/signal_racesys.c.o] Error 1
	make[1]: *** [suite/tests/CMakeFiles/linux.signal_racesys.dir/all] Error 2
	make: *** [all] Error 2
release-external-32: **** 4 build errors ****
	CMake Error at /.../dynamorio/core/CMake_readelf.cmake:55 (message):
	make[2]: *** [lib32/release/libdynamorio.so] Error 1
```

```
CMake Error at /work/dr/git/src/core/CMake_readelf.cmake:55 (message):
  *** Error:
  /work/dr/git/build_clang_latest_x86_rel/lib32/release/libdynamorio.so has
  text relocations
```

--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -87,6 +87,9 @@ add_library(drhelper STATIC ${shared_core_asm_src} ${shared_arch_asm_src}
   lib/dr_helper.c)
 # CMake on Windows fails to link unless we tell it to use the C linker:
 set_target_properties(drhelper PROPERTIES LINKER_LANGUAGE C)
+if (UNIX)
+  append_property_string(TARGET drhelper COMPILE_FLAGS "-fPIC")
+endif ()
 add_gen_events_deps(drhelper)
 if (WIN32 AND "${CMAKE_GENERATOR}" MATCHES "Visual Studio")
   # for parallel build correctness we need a target dependence

